RecastDemo/Source/ConvexVolumeTool.cpp

```diff
@@ -130,14 +130,18 @@ void ConvexVolumeTool::handleMenu()
 
 	imguiLabel("Area Type");
 	imguiIndent();
-	if (imguiCheck("Grass", m_areaType == SAMPLE_POLYAREA_GRASS))
-		m_areaType = SAMPLE_POLYAREA_GRASS;
-	if (imguiCheck("Road", m_areaType == SAMPLE_POLYAREA_ROAD))
-		m_areaType = SAMPLE_POLYAREA_ROAD;
+// 	if (imguiCheck("Ground", m_areaType == SAMPLE_POLYAREA_GROUND))
+// 		m_areaType = SAMPLE_POLYAREA_GROUND;
```

Comment:
Yes you are right about the holes.

The reasoning for this is that area types change meaning in the build process.
Until regions are built there is a special flag to indicate holes - this flag is `RC_NULL_AREA` which as you point out coincides with `SAMPLE_POLYAREA_GROUND`.

After regions have been built this flag is no longer meaningful, and Recast can support the entire range of areas (64 different areas). The remapping is handled [here.](https://github.com/recastnavigation/recastnavigation/blob/ad6e5de2e9e775fc06c20375b3bf64ee8ad26b06/RecastDemo/Source/Sample_TileMesh.cpp#L1231)

The original source of the `RC_WALKABLE_AREA` flag is [`rcMarkWalkableTriangles`.](https://github.com/recastnavigation/recastnavigation/blob/ad6e5de2e9e775fc06c20375b3bf64ee8ad26b06/Recast/Source/Recast.cpp#L281)
This function is used in the sample.

So several options for you:
1. Leave it alone for now and uncomment it again. It can be fixed in a future PR.
2. Resolve it by remapping `SAMPLE_POLYAREA_GROUND` to `RC_WALKABLE_AREA` during convex poly marking

The ideal solution (which should happen separately from these changes) is something like:
`RC_WALKABLE_AREA` is not meaningful to the build process in any way, it is simply a flag with value 63. `rcMarkWalkableTriangles` should be changed to accept the flag to use for marking the walkable triangles instead of always using flag 63. After this change all the `SAMPLE_POLYAREA` values should have 1 added to them (or implicitly add 1 during rasterization and subtract 1 in the poly mesh), and `SAMPLE_POLYAREA_GROUND` should be passed to `rcMarkWalkableTriangles`.

I don't expect you to submit a PR for this last bit. If you don't I will probably do it some time in the future when I have time, but please do either 1 or 2.

Anyway thanks for the work!